goldrush_path/goldrush_path.cpp

```diff
@@ -34,6 +35,27 @@
 #include <tuple>
 #include <vector>
 
+double sum_phred(const std::string& qual)
+{
+  double phred_sum = 0;
+  for (const auto& phred : qual) {
+    int phred_score = (int)(phred - 33);
+
+    // Delog the Phred score: 10^(-Q/10)
+    double delog_phred = pow(10.0, -phred_score / 10.0);
+
+    phred_sum += delog_phred;
+  }
+  return phred_sum;
+}

```

Comment:
This code seems to be duplicate here, and within `calc_phred_average.cpp` - Could we consolidate that? I realize that the latter also does the two halves, but still worth thinking about to reduce code duplication.
Also, seems like some of the code for dealing with phred scores is in `calc_phred_average.cpp` and some is in this file - would it make more sense to organize all the phred calculation code in a single file?

Reply:
I agree and right now my quick plan for the future is to simply test if the d parameter does much. If not, I will remove it, otherwise I will refactor it out. For the purpose of this PR, I don't want to change this since I'll be tackling it later.

Reply:
Fair enough! At minimum though - would it not make sense for this code to be in `calc_phred_average.cpp`? Just worried that having duplicated code plus being across multiple different files will be easily forgotten about in the future, etc.

Reply:
done